relations/src/environment.rs

```diff
@@ -0,0 +1,139 @@
+use ark_poly::univariate::DensePolynomial;
+use ark_poly_commit::marlin_pc::MarlinKZG10;
+use ark_relations::r1cs::ConstraintSynthesizer;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use ark_snark::SNARK;
+use ark_std::rand::{rngs::StdRng, SeedableRng};
+use blake2::Blake2s;
+
+// For now, we can settle with these types.
+/// Common pairing engine.
+pub type PairingEngine = ark_bls12_381::Bls12_381;
+/// Common scalar field.
+pub type CircuitField = ark_bls12_381::Fr;
+
+// Systems with hardcoded parameters.
+pub type Groth16 = ark_groth16::Groth16<PairingEngine>;
+pub type GM17 = ark_gm17::GM17<PairingEngine>;
+pub type MarlinPolynomialCommitment = MarlinKZG10<PairingEngine, DensePolynomial<CircuitField>>;
+pub type Marlin = ark_marlin::Marlin<CircuitField, MarlinPolynomialCommitment, Blake2s>;
+
+/// Serialized keys.
+pub struct RawKeys {
+    pub pk: Vec<u8>,
+    pub vk: Vec<u8>,
+}
+
+/// Common API for every proving system.
+pub trait ProvingSystem {
```

Comment:
I'm wondering whether this crate should also allow to verify proofs... wdyt?

Reply:
very good idea, pushed it